HousingSearchApi.Tests/V2/E2ETests/Fixtures/ElasticsearchFixture.cs

```diff
@@ -21,49 +21,61 @@ public ElasticsearchFixture()
-        var client = new ElasticClient();
+        var existsResponse = Client.Indices.Exists(indexName);
+        if (existsResponse.Exists)
+            return;
 
-        var response = await client.LowLevel.Indices.CreateAsync<StringResponse>(indexName, indexDefinition);
+        var indexDefinition = File.ReadAllText(Path.Combine(_indexFilesPath, filename));
+        var response = Client.LowLevel.Indices.Create<StringResponse>(indexName, indexDefinition);
 
         if (!response.Success)
-            Console.WriteLine($"Failed to create index: {response.DebugInformation}");
-        else
-            Console.WriteLine("Index created successfully.");
+            throw new Exception("Failed to create index: " + response.DebugInformation);
+        Console.WriteLine("Index created successfully.");
     }
 
-    public async Task LoadDataAsync(string filename)
+    public void LoadData(string filename, string indexName)
     {
+        Console.WriteLine($"Loading data from {filename} into index {indexName}");
         string jsonFilePath = Path.Combine(FixtureFilesPath, filename);
         if (!File.Exists(jsonFilePath))
             throw new FileNotFoundException($"The file {jsonFilePath} could not be found in directory {Directory.GetCurrentDirectory()}");
 
+        string jsonContent = File.ReadAllText(jsonFilePath);
 
-        string jsonContent = await File.ReadAllTextAsync(jsonFilePath);
-
-        var bulkResponse = await Client.LowLevel.BulkAsync<StringResponse>(PostData.String(jsonContent));
+        // Perform the bulk insert with immediate refresh
+        var bulkResponse = Client.LowLevel.Bulk<StringResponse>(
+            PostData.String(jsonContent),
+            new BulkRequestParameters { Refresh = Refresh.WaitFor }
+        );
 
         if (!bulkResponse.Success)
             throw new Exception("Bulk insert failed: " + bulkResponse.DebugInformation);
-
     }
 
-    public async Task InitializeAsync()
+    public Task InitializeAsync()
     {
         var indexSettingsFiles = new string[] { "assetIndex.json", "tenureIndex.json", "personIndex.json" };
+
         foreach (string filename in indexSettingsFiles)
-            await CreateIndexAsync(filename, indexName: filename.Replace("Index.json", ""));
+        {
+            var indexName = filename.Replace("Index.json", "") + "s";
+            CreateIndex(filename, indexName);
+        }
 
         var filenames = new string[] { "assets.json", "tenures.json", "persons.json" };
         foreach (string filename in filenames)
-            await LoadDataAsync(filename);
+        {
+            var indexName = filename.Replace("Index.json", "") + "s";
+            LoadData(filename, indexName);
+        }
+
+        return Task.CompletedTask;
     }
 
     public Task DisposeAsync()
     {
-        // Clean up if necessary, e.g., deleting the index
         return Task.CompletedTask;
     }
 }
```

Comment:
Was getting race conditions where the tests would run before the data finished loading. Making these synchronous and waiting for completion seems to solve that without increasing test duration